src/dos/program_mem.cpp

```diff
@@ -83,6 +94,12 @@ void MEM::Run(void)
 		}
 	}
 
+	// Free the base and UMB allocations made above
+	while (!allocated_segs.empty()) {
+		DOS_FreeMemory(allocated_segs.top());
+		allocated_segs.pop();
+	}
+
```

Comment:
Huh? They _allocated_ memory to check how much memory is free, then didn't free it? 🤦🏻 Can't we figure out how much free memory there is without _allocating_ it? 🤣

Reply:
It's probably easiest way to gather MEM's stats; like total number of blocks, largest chunk, and combined amount. 

MEM sits outside the "kernel" so also isn't privy to internals.. and without calls that give back these values, MEM would have to scan the memory for the memory chain block structure and so on (which would re-implement a lot of what DOS internals do).

Reply:
Kinda makes sense, okay... But... man... You think you've seen everything, then this... 😐

Reply:
Yup.. FreeDOS's MEM sources inspect the data structures instead allocating, which weighs in just under 4000 LOC across two files:
- https://gitlab.com/FreeDOS/base/mem/-/blob/master/SOURCE/MEM/MEM.C
- https://gitlab.com/FreeDOS/base/mem/-/blob/master/SOURCE/MEM/MEM2.C

Reply:
I think this could screw up TSRs perhaphs that do try to allocate memory while MEM is doing it's weird alloc the world the free it dance... Probably low priority, but just saying. You could probably cause interesting failures in those DOS multitasking environments too by running some memory intensive program while issuing a few MEM commands simultaneously. Ugh...